### PR TITLE
Add PyTorch ImageNet example

### PR DESCRIPTION
**Pull request by alsrgv**
Converges to 76%+ Top-1 accuracy on 128 GPUs.

### CODE REVIEW

**Comment by bapriddy** [comment]
Just wanted to let you know it runs without errors although I had to comment out the TensorboardX stuff.  Nice work!!  I was in the process of converting Soumith's version to use with Horovod but you guys beat me.  Thanks!  I'm hoping to out-perform G**gle's tf.

**Comment by alsrgv** [comment]
@bapriddy, thanks!  `tensorboardX` is very useful, you can install it via `pip install tensorboardX`.

**Comment by bapriddy** [comment]
in the line
'''kwargs = {'num_workers': 4, 'pin_memory': True} if args.cuda else {}'''
if i'm running on say 8 or 16 nodes, each node is a mother + cpu + gpu, should 'num workers'.  according to the notes thses are dataloaders but not sure about the details.

Second question...In the out put is it/s iterations per second...?
